test/integration/routes/metadata/blob-reference.test.js

```diff
@@ -0,0 +1,112 @@
+const ANOTHER_MOCK_CORRELATION_ID = 'another-mock-correlation-id'
+const MOCK_BLOB_REFERENCE = mockEvent.metadata.data.blobReference
+const MOCK_EVENT = mockEvent.metadata
+
+let server
+
+beforeAll(async () => {
+  server = await startServer()
+})
+
+beforeEach(async () => {
+  if (!db.client.topology?.isConnected()) {
+    await db.client.connect()
+  }
+  await clearCollection(METADATA_COLLECTION)
+})
+
+afterAll(async () => {
+  await server.stop()
+  await db.client.close()
+})
+
+describe('GET /api/v1/metadata/events/blob-reference/{blobReference}', () => {
+  test('Return 200 when single document is found with corresponding blobReference', async () => {
+    await insertMockEventToDb(METADATA_COLLECTION, MOCK_CORRELATION_ID, MOCK_EVENT)
+
+    const response = await makeApiRequest(server, BASE_URL, MOCK_BLOB_REFERENCE)
+
+    expect(response.statusCode).toBe(200)
+    expect(response.headers['content-type']).toContain('application/json')
+    expect(response.payload).toEqual(JSON.stringify({
+      data: [{
+        correlationId: MOCK_CORRELATION_ID,
+        events: [MOCK_EVENT]
+      }]
+    }))
+  })
+  test('Return 200 when multiple documents are found with corresponding blobReference', async () => {
+    await insertMockEventToDb(METADATA_COLLECTION, MOCK_CORRELATION_ID, MOCK_EVENT)
+    await insertMockEventToDb(METADATA_COLLECTION, ANOTHER_MOCK_CORRELATION_ID, MOCK_EVENT)
+
+    const response = await makeApiRequest(server, BASE_URL, MOCK_BLOB_REFERENCE)
+
+    expect(response.statusCode).toBe(200)
+    expect(response.headers['content-type']).toContain('application/json')
+    expect(response.payload).toEqual(JSON.stringify({
+      data: [{
+        correlationId: MOCK_CORRELATION_ID,
+        events: [MOCK_EVENT]
+      },
+      {
+        correlationId: ANOTHER_MOCK_CORRELATION_ID,
+        events: [MOCK_EVENT]
+      }]
+    }))
+  })
+
+  test('Return 404 when no documents are found with corresponding blobReference', async () => {
+    const nonExistingBlobReference = 'not-found-blob-reference'
+
+    const response = await makeApiRequest(server, BASE_URL, nonExistingBlobReference)
+
+    expect(response.statusCode).toBe(404)
+    expect(response.headers['content-type']).toContain('application/json')
+    expect(response.payload).toEqual(JSON.stringify({
+      statusCode: 404,
+      error: 'Not Found',
+      message: `No document found for the provided blob reference: ${nonExistingBlobReference}`
+    }))
+  })
+  test('Return 400 when blobReference is not provided', async () => {
+    const invalidBlobReference = '%invalid-reference'
+
+    const response = await makeApiRequest(server, BASE_URL, invalidBlobReference)
+
+    expect(response.statusCode).toBe(400)
+    expect(response.headers['content-type']).toContain('application/json')
+    expect(response.payload).toEqual(JSON.stringify({
+      statusCode: 400,
+      error: 'Bad Request',
+      message: 'Bad Request'
```

Comment:
Should we be providing more detail than this?

Reply:
Actually this might be down to our vague validation rules so one for the future once we know more about what the blob reference format will be